docs/download.md

```diff
@@ -49,6 +49,7 @@ The plugin is tested on the following architectures:
 		Databricks 11.3 ML LTS (GPU, Scala 2.12, Spark 3.3.0)
 		Databricks 12.2 ML LTS (GPU, Scala 2.12, Spark 3.3.2)
 		Databricks 13.3 ML LTS (GPU, Scala 2.12, Spark 3.4.1)
+		Databricks 14.3 ML LTS (GPU, Scala 2.12, Spark 3.5.0)
```

Comment:
We should not announce to support DB14.3, it's just an experimental shim for user?  @sameerz  please help double confirm, thanks

Reply:
removed, thx